src/toolchain.rs

```diff
@@ -123,7 +125,7 @@ impl<'a> Toolchain<'a> {
             (self.cfg.notify_handler)(Notification::InstalledToolchain(&self.name));
         }
 
-        let status = match (updated, exists) {
+        let status = match (updated, previous_version.is_some()) {
             (true, false) => UpdateStatus::Installed,
             (true, true) => UpdateStatus::Updated(previous_version.unwrap()),
             (false, true) => UpdateStatus::Unchanged,
```

Comment:
This match block can be simplified into something like:

```rust
let status = match (updated, previous_version) {
    (true, None) => UpdateStatus::Installed,
    (true, Some(v)) => UpdateStatus::Updated(v),
    (false, _) => UpdateStatus::Unchanged,
};
```

Perhaps?

Reply:
Yes, that's much clearer.